src/components/event/EventForm.js

```diff
@@ -0,0 +1,95 @@
+
+    /*
+        Since the input fields are bound to the values of
+        the properties of this state variable, you need to
+        provide some default values.
+    */
+    const [currentEvent, setCurrentEvent] = useState({
+        gameId: 0,
+        description: "",
+        date: "",
+        time: "",
+        organizer: 0
+    })
+
+    useEffect(() => {
+        getGames().then(data => setGames(data))
+    }, [])
+
+    const changeEventState = (evt) => {
+        let stateEvent = { ...currentEvent }
+        stateEvent[evt.target.name] = evt.target.value
+        setCurrentEvent(stateEvent)
+    }
+
+    return (
+        <form className="eventForm">
+        <h2 className="eventForm_Title">Register New Event</h2>
+        <fieldset>
+            <div className="form-group">
+                <label htmlFor="game">Game: </label>
+                <select type="text" name="gameId" required autoFocus className="form-control"
+                    value={parseInt(currentEvent.gameId)}
+                    onChange={changeEventState}>
+                        <option value={0}>Select a Game</option>
+                        {
+                            games.map(game => {
+                                return <option value={game.id}>{game.title}</option>
+                            })
+                        }
+                    </select>
+            </div>
+            <div className="form-group">
+                <label htmlFor="description">Description: </label>
+                <input type="text" name="description" required className="form-control"
+                    value={currentEvent.description}
+                    onChange={changeEventState}
+                />
+            </div>
+            <div className="form-group">
+                <label htmlFor="date">Date: </label>
+                <input type="date" name="date" required className="form-control"
+                    value={currentEvent.date}
+                    onChange={changeEventState}
+                />
+            </div>
+            <div className="form-group">
+                <label htmlFor="time">Time: </label>
+                <input type="time" name="time" required className="form-control"
+                    value={currentEvent.time}
+                    onChange={changeEventState}
+                />
+            </div>
+        </fieldset>
+
+        {/* TODO: create the rest of the input fields */}
+
+        <button type="submit"
+            onClick={evt => {
+                // Prevent form from being submitted
+                evt.preventDefault()
+
+                const event = {
+                    game: currentEvent.gameId,
+                    description: currentEvent.description,
+                    date: currentEvent.date,
+                    time: currentEvent.time
+                }
+
+                // Send POST request to your API
+                createEvent(event)
+                    .then(() => navigate("/events"))
```

Comment:
This is a good use of `navigate` because this is not happening as a result of a link click. This is a side effect of successfully creating an event.

Reply:
Once again, what about error handling?